netmiko/nokia/nokia_srl.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2014 - 2022 Kirk Byers
+# Copyright (c) 2014 - 2022 Twin Bridges Technology
+# Copyright (c) 2019 - 2020 NOKIA Inc.
+# MIT License - See License file at:
+#   https://github.com/ktbyers/netmiko/blob/develop/LICENSE
+
+import re
+import time
+from typing import Any, Optional, Sequence, TextIO, Union
+from netmiko import log
+from netmiko.no_enable import NoEnable
+from netmiko.base_connection import BaseConnection
+
+
+class NokiaSrlSSH(BaseConnection, NoEnable):
+    """
+    Implement methods for interacting with Nokia SRL devices for SSH.
+
+    Not applicable in Nokia SRL:
+        - check_enable_mode()
+        - enable()
+        - exit_enable_mode()
+
+    Overriden methods to adapt Nokia SR OS behavior (changed):
+        - session_preparation()
+        - set_base_prompt()
+        - config_mode()
+        - exit_config_mode()
+        - check_config_mode()
+        - save_config()
+        - commit()
+        - strip_prompt()
+        - enable()
+        - check_enable_mode()
+
+    By default, the SR Linux CLI prompt consists of two lines of text,
+    indicating with an asterisk whether the configuration has been modified or
+    a plus sign whether the configuration has been saved, the current mode and
+    session type, the current CLI context, and the host name of the SR Linux device.
+
+    Examples:
+
+    --{ running }--[ interface ethernet-1/1 subinterface 1 ]--
+    A:ams01#
+
+    --{ * candidate private private-admin }--[ interface ethernet-1/1 subinterface 1 ]--
+    A:ams01#
+
+    --{ + candidate private private-admin }--[ interface ethernet-1/1 subinterface 1 ]--
+    A:ams01#
+
+    This class support the default prompt configuration.
+    """
+
+    def session_preparation(self) -> None:
+        self._test_channel_read()
+        self.ansi_escape_codes = True
+        # Bottom toolbar text not required
+        command = "environment cli-engine type basic"
+        self.write_channel(self.normalize_cmd(command))
+        command = "environment complete-on-space false"
+        self.write_channel(self.normalize_cmd(command))
+        self.set_base_prompt()
+        time.sleep(10 * self.global_delay_factor)
+        self.clear_buffer()
```

Comment:
Next thing tot appears is just the prompt

Reply:
Okay, it would be better for the performance of this driver to do the following:

```python
        # Overloading disable_paging which is confusing
        command = "environment cli-engine type basic"
        self.disable_paging(
            command=command,
            pattern=r"#",
        )   
        command = "environment complete-on-space false"
        self.disable_paging(
            command=command, pattern=r"#"
        )  
```

And get rid of the sleep and clear_buffer call.

This is a bit of kludge using disable_paging, but it is fine, I have used it elsewhere (I am just using it to send command and search for a pattern and also for command echo verification).

Let me know if that doesn't work though.